archive/imo/imo1969_q1.lean

```diff
@@ -0,0 +1,81 @@
+import tactic
+import tactic.basic
+import tactic.linarith
+import tactic.norm_cast
+import tactic.ring
+
+open int
+open nat
+
+/-
+The 1969 IMO, problem 1:
+Prove that there are infinitely many natural numbers $a$ with the following property:
+the number $z = n^4 + a$ is not prime for any natural number $n$.
+
+The key to the solution is that you can factor this into the product of two polynomials, if a = 4*m^4.
+-/
+
+lemma factorization (m n: ℤ): (n^2 + 2*m^2 - 2*n*m) * (n^2 + 2*m^2 + 2*n*m) = n^4 + 4*m^4
+:= by ring
+
+/-
+To show that the product is not prime, we need to show each of the factors is at least 2, which we can
+do with a sum-of-squares expression.
```

Comment:
Although we don't currently display `archive/` files on our website, when we do, enclosing comments in `/-!` and `-/` will mark them as module doc strings and cause them to be displayed. 

Also we try to keep lines < 100 characters in length.
```suggestion
/-!
To show that the product is not prime, we need to show each of the factors is at least 2, which we
can do with a sum-of-squares expression.
```

Reply:
ok